### PR TITLE
[onert/core] Fix ShapeValidator of RmsNorm

### PR DESCRIPTION
**Pull request by seockho-kim**
This commit fixes ShapeValidator of RmsNorm to allow rank 3 input and single gamma.

ONE-DCO-1.0-Signed-off-by: Seockho Kim seockho.kim@samsung.com

issue: https://github.com/Samsung/ONE/issues/14089
draft: https://github.com/Samsung/ONE/pull/14088

### CODE REVIEW

File: runtime/onert/core/src/compiler/ShapeValidator.cc [hunk 1]
```diff
@@ -1133,9 +1133,15 @@ void ShapeValidator::visit(const ir::operation::RmsNorm &node)
   const auto ifm_index{node.getInputs().at(ir::operation::RmsNorm::Input::INPUT)};
   const auto gamma_index{node.getInputs().at(ir::operation::RmsNorm::Input::GAMMA)};
 
-  OP_REQUIRES(operands.at(ifm_index).shape().rank() == 4);
-  OP_REQUIRES(operands.at(ifm_index).shape() == operands.at(ofm_index).shape());
-  OP_REQUIRES(operands.at(gamma_index).shape().rank() == 1);
+  auto ifm_shape = operands.at(ifm_index).shape();
+  auto ofm_shape = operands.at(ofm_index).shape();
+  auto gamma_shape = operands.at(gamma_index).shape();
```
Comment (zetwhite) [inline runtime/onert/core/src/compiler/ShapeValidator.cc:1138]: (optional) How about adding `const reference`?  

```suggestion
  const auto& ifm_shape = operands.at(ifm_index).shape();
  const auto& ofm_shape = operands.at(ofm_index).shape();
  const auto& gamma_shape = operands.at(gamma_index).shape();
```

File: runtime/onert/core/src/compiler/ShapeValidator.cc [hunk 2]
```diff
@@ -1133,9 +1133,15 @@ void ShapeValidator::visit(const ir::operation::RmsNorm &node)
...
+
+  OP_REQUIRES(ifm_shape.rank() == 3 || ifm_shape.rank() == 4);
```
Comment (zetwhite) [inline runtime/onert/core/src/compiler/ShapeValidator.cc:1140]: IDK the background, Is there a reason to support input with `rank==3`?

**Review by zetwhite — DISMISSED** [review]
LGTM

File: runtime/onert/core/src/compiler/ShapeValidator.cc (hunk 2 above)
Comment (seockho-kim) [reply runtime/onert/core/src/compiler/ShapeValidator.cc:1140]: > IDK the background, Is there a reason to support input with `rank==3`?

Current testing model has "rank = 3" inputs, like A x B x C.

File: runtime/onert/core/src/compiler/ShapeValidator.cc (hunk 1 above)
Comment (seockho-kim) [reply runtime/onert/core/src/compiler/ShapeValidator.cc:1138]: > (optional) How about adding `const reference`?

Okay, I'll update it :)

**Review by glistening — APPROVED** [review]
LGTM

**Review by zetwhite — APPROVED** [review]

**Pull request merged**